### PR TITLE
Fix bug where prediction explanations class_name was shown as float for boolean targets

### LINKED ISSUE #3178: Prediction Explanations shows boolean class name as numeric in "class_name" field
Repro

```python
from evalml.demos import load_fraud
from evalml.pipelines import BinaryClassificationPipeline

X, y = load_fraud(1000)

pl = BinaryClassificationPipeline({'Label Encoder': ['Label Encoder', 'X', 'y'],
                                   'DateTime Featurization Component': ['DateTime Featurization Component', 'X', 'Label Encoder.y'],
                                   'Imputer': ['Imputer', 'DateTime Featurization Component.x', 'Label Encoder.y'],
                                   'One Hot Encoder': ['One Hot Encoder', 'Imputer.x', 'Label Encoder.y'],
                                   'Oversampler': ['Oversampler', 'One Hot Encoder.x', 'Label Encoder.y'],
                                   'Standard Scaler': ['Standard Scaler', 'Oversampler.x', 'Oversampler.y'],
                                   'Elastic Net Classifier': ['Elastic Net Classifier', 'Standard Scaler.x', 'Oversampler.y']})
pl.fit(X, y)

exp = explain_predictions_best_worst(pl, X, y, output_format="dict")
assert exp['explanations'][0]['explanations'][0]['class_name'] == 1.0
```

The `class_name` field should be boolean since the labels in the original data are boolean.

### PR DESCRIPTION
**Pull request by freddyaboulton**
### Pull Request Description
Fixes #3178

-----
*After creating the pull request: in order to pass the **release_notes_updated** check you will need to update the "Future Release" section of* `docs/source/release_notes.rst` *to include this pull request by adding :pr:`123`.*

### CODE REVIEW

File: evalml/tests/model_understanding_tests/prediction_explanations_tests/test_explainers.py [hunk 1]
```diff
@@ -1759,6 +1759,60 @@ def test_explain_predictions_oversampler(estimator, algorithm, fraud_100):
...
+        pipeline,
+        X,
+        y,
+        num_to_explain=1,
+        output_format="dict",
+        top_k_features=2,
+        algorithm="shap",
+    )
+    if problem_type == ProblemTypes.BINARY:
+        assert exp["explanations"][0]["explanations"][0]["class_name"]
```
Comment (jeremyliweishih) [inline evalml/tests/model_understanding_tests/prediction_explanations_tests/test_explainers.py:1807]: Should we also explicitly assert that the class name is a boolean?

**Review by jeremyliweishih — APPROVED** [review]
LGTM but left a comment about explicitly asserting that the `class_name` field is a boolean.

File: evalml/tests/model_understanding_tests/prediction_explanations_tests/test_explainers.py (hunk 1 above)
Comment (freddyaboulton) [reply evalml/tests/model_understanding_tests/prediction_explanations_tests/test_explainers.py]: Great call! Pushing this up now.

File: evalml/tests/model_understanding_tests/prediction_explanations_tests/test_explainers.py [hunk 2]
```diff
@@ -1759,6 +1759,61 @@ def test_explain_predictions_oversampler(estimator, algorithm, fraud_100):
     assert report["feature_values"].isnull().sum() == 0
 
 
+@pytest.mark.noncore_dependency
+@pytest.mark.parametrize("problem_type", [ProblemTypes.MULTICLASS, ProblemTypes.BINARY])
+def test_explain_predictions_class_name_matches_class_name_in_y(
+    problem_type, fraud_100
+):
+    if problem_type == ProblemTypes.BINARY:
+        X, y = fraud_100
```
Comment (angela97lin) [inline evalml/tests/model_understanding_tests/prediction_explanations_tests/test_explainers.py:1768]: omega-nit-pick but `X, y = fraud_100` is a duplicate line for both conditions and can be moved out? :)

**Review by angela97lin — APPROVED** [review]
LGTM, thanks for doing this @freddyaboulton 👍

**Review by chukarsten — APPROVED** [review]
Looks good!

File: docs/source/release_notes.rst [hunk 3]
```diff
@@ -5,6 +5,7 @@ Release Notes
         * Added string support for DataCheckActionCode :pr:`3167`
         * Added ``DataCheckActionOption`` class :pr:`3134`
     * Fixes
+        * Fix bug where prediction explanations ``class_name`` was shown as float for boolean targets :pr:`3179`
```
Comment (eccabay) [inline docs/source/release_notes.rst:8]: Nit: Fix -> Fixed

**Pull request merged**